ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1172,8 +1215,22 @@ private StoreFindToken findEntriesFromSegmentStartOffset(Offset initialSegmentSt
    * @param messageEntries The message entry list where duplicates need to be removed
    */
   private void eliminateDuplicates(List<MessageInfo> messageEntries) {
```

Comment:
Same idea with `eliminateDuplicates(indexEntries)`. 
Is it possible to do `updateStateForMessages()` + `eliminateDuplicates(messageEntries)` inside caller's loop?

This looks like not a easy change. Maybe we can do it in a new PR as enhancement. 

----
Question: What's the plan for TTL replication? Replicate TTL entries to all peer nodes?

Reply:
Yeah that would be harder to do.

Yup either replicate the record or change the blob properties as the data is received for replication in `MessageSievingInputStream`